pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1041,7 +1041,16 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPListenerOptsForPor
 	if pluginParams.Port.Port == 0 {
 		rdsName = listenerOpts.bind // use the UDS as a rds name
 	} else {
-		rdsName = fmt.Sprintf("%d", pluginParams.Port.Port)
+		if pluginParams.ListenerProtocol == plugin.ListenerProtocolAuto &&
+			util.IsProtocolSniffingEnabledForNode(node) && listenerOpts.bind != actualWildcard {
+			if pluginParams.Service != nil {
+				rdsName = fmt.Sprintf("%s:%d", pluginParams.Service.Hostname, pluginParams.Port.Port)
+			} else {
+				rdsName = fmt.Sprintf("%s:%d", listenerOpts.bind, pluginParams.Port.Port)
```

Comment:
?? what is this for? this will end up returning empty RDS when the bind is an IP declared by the Sidecar.